dataprep/eda/correlation/compute/bivariate.py

```diff
@@ -55,51 +37,45 @@ def _calc_bivariate(
         labels[infidx[-k:]] = "-"  # type: ignore
         # pylint: enable=invalid-unary-operand-type
         labels[infidx[:k]] = "+"
-        result["data"]["influence"] = labels
+        result["data"]["influence"] = labels  # type: ignore
 
     return Intermediate(**result, visual_type="correlation_scatter")
 
 
 def scatter_with_regression(
-    x: da.Array, y: da.Array, sample_size: int, k: Optional[int] = None
+    df: dd.DataFrame, sample_size: int, k: Optional[int] = None
 ) -> Tuple[Tuple[da.Array, da.Array], Tuple[da.Array, da.Array], Optional[da.Array]]:
     """Calculate pearson correlation on 2 given arrays.
 
     Parameters
     ----------
-    xarr : da.Array
-    yarr : da.Array
-    sample_size : int
+    df
+        dataframe
+    sample_size
+        Number of points to show in the scatter plot
     k : Optional[int] = None
         Highlight k points which influence pearson correlation most
     """
-    if k == 0:
-        raise ValueError("k should be larger than 0")
-
-    xp1 = da.vstack([x, da.ones_like(x)]).T
-    xp1 = xp1.rechunk((xp1.chunks[0], -1))
-
-    mask = ~(da.isnan(x) | da.isnan(y))
-    # if chunk size in the first dimension is 1, lstsq will use sfqr instead of tsqr,
-    # where the former does not support nan in shape.
-
-    if len(xp1.chunks[0]) == 1:
-        xp1 = xp1.rechunk((2, -1))
-        y = y.rechunk((2, -1))
-        mask = mask.rechunk((2, -1))
-
-    (coeffa, coeffb), _, _, _ = da.linalg.lstsq(xp1[mask], y[mask])
-
-    if sample_size < x.shape[0]:
-        samplesel = da.random.choice(x.shape[0], int(sample_size), chunks=x.chunksize)
-        x = x[samplesel]
-        y = y[samplesel]
-
-    if k is None:
-        return (coeffa, coeffb), (x, y), None
+    df["ones"] = 1
+    arr = df.to_dask_array(lengths=True)
+
+    (coeffa, coeffb), _, _, _ = da.linalg.lstsq(arr[:, [0, 2]], arr[:, 1])
+
+    df = df.drop(columns=["ones"])
+    df_smp = df.map_partitions(
+        lambda x: x.sample(min(sample_size, x.shape[0])), meta=df
+    )
+    # TODO influences should not be computed on a sample
+    influences = (
+        pearson_influence(
+            df_smp[df.columns[0]].to_dask_array(lengths=True),
```

Comment:
These two lines will trigger computes right?

Reply:
yes

Reply:
Got it. I think we can optimize it using graph later.